src/mame/jaleco/megasys1.cpp

```diff
@@ -5238,105 +5274,44 @@ void megasys1_bc_iosim_state::init_cybattlr()
 	save_item(NAME(m_sprite_bank));
 }
 
-void megasys1_bc_iosim_state::init_edf()
+void megasys1_bc_iosim_state::init_chimeraba() // Type C
 {
-	m_ip_select_values[0] = 0x20;
-	m_ip_select_values[1] = 0x21;
-	m_ip_select_values[2] = 0x22;
-	m_ip_select_values[3] = 0x23;
-	m_ip_select_values[4] = 0x24;
+	m_ip_select_values[0] = 0x56;
+	m_ip_select_values[1] = 0x52;
+	m_ip_select_values[2] = 0x53;
+	m_ip_select_values[3] = 0x55;
+	m_ip_select_values[4] = 0x54;
 
-	m_ip_select_values[5] = 0xf0;
+	m_ip_select_values[5] = 0xfa;
 	m_ip_select_values[6] = 0x06;
 
 	save_item(NAME(m_ip_latched));
+	save_item(NAME(m_sprite_bank));
 }
 
-void megasys1_state::init_edfp()
-{
-	phantasm_rom_decode(machine(), "maincpu");
-}
-
-void megasys1_bc_iosim_state::init_hayaosi1()
+void megasys1_bc_iosim_state::init_cybattlr() // Type C
 {
-	m_ip_select_values[0] = 0x51;
+	m_ip_select_values[0] = 0x56;
 	m_ip_select_values[1] = 0x52;
 	m_ip_select_values[2] = 0x53;
 	m_ip_select_values[3] = 0x54;
 	m_ip_select_values[4] = 0x55;
 
-	m_ip_select_values[5] = 0xfc;
+	m_ip_select_values[5] = 0xf2;
 	m_ip_select_values[6] = 0x06;
 
 	save_item(NAME(m_ip_latched));
+	save_item(NAME(m_sprite_bank));
```

Comment:
Can these be saved in `machine_start` rather than repeating it in all the init functions for the class that simulates the microcontroller?

Reply:
the ip_latched thing can be, for sprite_bank seems to be a 1c specific feature that might get tidied up later